### PR TITLE
CSCQVAIN-261: Remove jwt token from the front-end

### PR DESCRIPTION
**Pull request by tahme**
Fetch session data from the back-end after login instead of parsing them from a token. Changes the front-end login redirect url from /token to /login, which also needs a change in the back-end.

### CODE REVIEW

File: src/auth/auth.js [hunk 1]
```diff
@@ -121,15 +76,13 @@ Auth.prototype.setUser = function(user) {
 	this._user = user
 }
 
-Auth.prototype.login = function(token) {
-	this.setUser(UserFromToken(token))
-
+Auth.prototype.login = async function() {
+	this.loading.state = true
+	await this.resumeSession(false) // fetch user session data
```
Comment (Kauhia) [inline src/auth/auth.js]: Is this correct? The login doesn't seem to log in?

**Review by Kauhia — COMMENTED** [review]
It is quite difficult to follow how the system works..

**Comment by tahme** [comment]
> It is quite difficult to follow how the system works..

Did some renaming and refactored the logic a bit. Should work the same way, but hopefully its more clear now.

File: src/auth/Login.vue [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+			this.handleLoggedIn()
+		} else if (await this.$auth.loadSession()) { // Fetch session data from the backend
+			this.handleLoggedIn()
+		} else {
+			this.handleLoginError("sessionerror")
+		}
+	},
+	methods: {
+		handleLoggedIn() {
+			this.$router.push({ name: 'datasets' })
```
Comment (Kauhia) [inline src/auth/Login.vue:29]: Will user loose data if session is refreshed and user is redirected to datasets?

File: src/auth/auth.js (hunk 1 above)
Comment (tahme) [reply src/auth/auth.js]: Auth.prototype.login is now in Auth.prototype.loadSession to make it clearer what it does.

**Review by Kauhia — APPROVED** [review]

**Review by Kauhia — APPROVED** [review]

File: src/auth/Login.vue (hunk 2 above)
Comment (tahme) [reply src/auth/Login.vue:29]: If an already logged in user somehow manages to enter /login (shouldn't happen unless they type the url manually), they lose data the same way as if they entered any other route (e.g. /userinfo).

**Pull request merged**